### DISCUSSION TITLE
Transpile Remix + all dependencies for certain browser target

### DISCUSSION THREAD

**Discussion by nickluger** [remix-run/remix#8024] [Proposals]
There's https://github.com/remix-run/remix/discussions/4769 to target the transpilation of Remix app code to a certain browser target, but the reality is:

- If you really want to support certain browsers, you need to make sure **ALL your code** supports those browsers. A single `??` (Nullish coalescing operator) in your bundle can crash the whole app, whether it comes from Remix, your application code, your dependencies or your dependencies' dependencies.
- In a monorepo, you need to transpile your own packages/dependencies anyway. You control what they are transpiled to, but...
- You don't control what ECMA-Script version your npm dependencies transpile to or what their dependencies transpile to.
- The only place where everything comes together is your Remix app. This is the only place that could enforce a certain JS-version/feature set.
- Polyfills is something else that can already be handled by using something like https://polyfill.io/.

Is it possible to enforce a certain JS version/feature set on my whole bundle? Anyone already doing that? Can you transpile the bundle after its initial build by Remix? Is there already a feature like that? (`serverDependenciesToBundle` sounds like it's only affecting the server-side bundle, not the client-side bundle).

**Comment by gustavopch** [comment]
As Remix is migrating to use Vite as its compiler, I believe this concern won't be within Remix's scope anymore, but let's see what the contributors have to say.
